### PR TITLE
feat: add `yalc` spec 

### PR DESCRIPTION
**Pull request by jakeols**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
feature

**What is the current behavior? (You can also link to an open issue here)**
n/a

**What is the new behavior (if this is a feature change)?**
Adds a new spec for [yalc](https://github.com/wclr/yalc). 

**Additional info:**
Been using (and loving) fig for a long time now -- figured I'd add a spec for a cli I've been using frequently.

### CODE REVIEW

File: src/yalc.ts [hunk 1]
```diff
@@ -0,0 +1,158 @@
+const completionSpec: Fig.Spec = {
+  name: "yalc",
+  description: "",
```
Comment (SeparateRecords) [inline src/yalc.ts:3]: Can you add a description here? Useful for https://fig.io/manual

File: src/yalc.ts [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+          name: "--no-sig",
+          description:
+            "Disable adding hash signature of all files when copying package content",
+        },
+        {
+          name: "--content",
+          description: "Show included files in the published package",
+        },
+        {
+          name: "-no-workspace-resolve",
```
Comment (SeparateRecords) [inline src/yalc.ts:29]: Should this have two dashes, eg. `--no-workspace-resolve`?

File: src/yalc.ts [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+      description:
+        "Publish your package to the store and propagate all changes to existing yalc package installations",
+    },
+    {
+      name: "add",
+      description: "Copy the current version from the store to your project",
+      args: {
+        name: "package",
+        description: "The package you want to publish",
+      },
```
Comment (SeparateRecords) [inline src/yalc.ts:45]: Is there a command we can use to generate the names of the packages dynamically?

Args have a `generators` prop that allows you to write code that generates suggestions dynamically 🙂

File: src/yalc.ts [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+        "Alterative to 'add', instead using local '.yalc' as symlink source",
+    },
+    {
+      name: "update",
+      description: "Update package(s)",
+      args: {
+        name: "package",
+        description: "The package to update",
+        isOptional: true,
+      },
```
Comment (SeparateRecords) [inline src/yalc.ts:77]: Same thing here, could we generate the suggestions for this?

File: src/yalc.ts [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+        },
+        {
+          name: "show",
+          description:
+            "Show all packages to which chosen package has been added",
+          options: [
+            {
+              name: "package",
+            },
+          ],
```
Comment (SeparateRecords) [inline src/yalc.ts:121]: Should these `options` be `args`?

**Review by clo4 — DISMISSED** [review]
Looking good! Not much to change here, thanks so much for contributing 😊

File: src/yalc.ts (hunk 1 above)
Comment (jakeols) [reply src/yalc.ts]: Yep, [d28b340](https://github.com/withfig/autocomplete/pull/1446/commits/d28b340f7a725d514f067d5cc888da3c04403c11)

File: src/yalc.ts (hunk 2 above)
Comment (jakeols) [reply src/yalc.ts]: Good catch, [d28b340](https://github.com/withfig/autocomplete/pull/1446/commits/d28b340f7a725d514f067d5cc888da3c04403c11)

File: src/yalc.ts (hunk 5 above)
Comment (jakeols) [reply src/yalc.ts]: Updated, [d28b340](https://github.com/withfig/autocomplete/pull/1446/commits/d28b340f7a725d514f067d5cc888da3c04403c11)

File: src/yalc.ts (hunk 3 above)
Comment (jakeols) [reply src/yalc.ts]: Let me dig into this a little bit -- the list of published packages are available within `~/.yalc/packages` but want to confirm the behavior is similar across different OS regardless of the linking method.

Comment (SeparateRecords) [reply src/yalc.ts]: Awesome. [The installed deno scripts generator](https://github.com/withfig/autocomplete/blob/f72c03bc76bf14a571e92b68d5e82b56b944bd80/src/deno/generators.ts#L353-L364) might be similar to what you want then, so take a look at that for a reference. Some other files with generators you might want to look at are `brew.ts` and `kill.ts`.

Comment (jakeols) [reply src/yalc.ts]: Got a bit busy recently but added the generators in [a2e91dd](https://github.com/withfig/autocomplete/pull/1446/commits/a2e91dd36f85b842416f2946325b51fa87bc440a). The generators for listing available packages defaults to appending a version (eg. `mypackage@1.0.0`) while removing does not as the `yalc.lock` only supports a single version, so version should be irrelevant.

File: src/yalc.ts [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+      })
+      .map((path) => ({
+        name: path,
+        icon: "📦",
+        description: path,
+      })),
+};
+
+const getRemovablePackages: Fig.Generator = {
+  script: "ls .yalc",
```
Comment (fedeci) [inline src/yalc.ts:23]:
```suggestion
  script: "command ls .yalc",
```

File: src/yalc.ts [hunk 7]
```diff
@@ -0,0 +1,194 @@
+const generatePackages: Fig.Generator = {
+  script: "\\find ~/.yalc/packages -maxdepth 4 -iname 'package.json'",
```
Comment (fedeci) [inline src/yalc.ts:2]:
```suggestion
	// TODO: use the same as for npm and yarn package.json reverse lookup
  script: "command find ~/.yalc/packages -maxdepth 4 -iname 'package.json'",
```

**Review by fedeci — CHANGES_REQUESTED** [review]
Thanks for your work @judofyr! Can you please sign the CLA, commit the changes suggested and we are ready to land this! Sorry for late response :(

**Comment by fedeci** [comment]
@recheck

**Comment by fedeci** [comment]
recheck

**Comment by jakeols** [comment]
I have read the CLA Document and I hereby sign the CLA

File: src/yalc.ts (hunk 6 above)
Comment (jakeols) [reply src/yalc.ts]: [cf6b8da](https://github.com/withfig/autocomplete/pull/1446/commits/cf6b8da6ee4d06a442dafa8dfc1ab18c77bdf25c)

File: src/yalc.ts (hunk 7 above)
Comment (jakeols) [reply src/yalc.ts]: [cf6b8da](https://github.com/withfig/autocomplete/pull/1446/commits/cf6b8da6ee4d06a442dafa8dfc1ab18c77bdf25c)

**Review by fedeci — APPROVED** [review]
Thanks!

**Pull request merged**